packages/web-worker/src/bugsnag.ts

```diff
@@ -0,0 +1,101 @@
+/* eslint-env worker, serviceworker */
+
+import delivery from '@bugsnag/delivery-fetch'
+import pluginClientIp from '@bugsnag/plugin-client-ip'
+import pluginWindowOnError from '@bugsnag/plugin-window-onerror'
+import pluginWindowUnhandledRejection from '@bugsnag/plugin-window-unhandled-rejection'
+
+// extend the base config schema with some browser-specific options
+import { schema as baseConfig } from '@bugsnag/core/config'
+import workerConfig from './config'
+import pluginBrowserDevice from '@bugsnag/plugin-browser-device'
+import pluginBrowserSession from '@bugsnag/plugin-browser-session'
+import pluginPreventDiscard from './prevent-discard'
+import ClientWithInternals from '@bugsnag/core/client'
+import type { Client, Config, BugsnagStatic } from '@bugsnag/core'
+import assign from '@bugsnag/core/lib/es-utils/assign'
+
+export interface WorkerConfig extends Config {
+  collectUserIp?: boolean
+  generateAnonymousId?: boolean
+}
+
+export interface WorkerBugsnagStatic extends BugsnagStatic {
+  start(apiKeyOrOpts: string | WorkerConfig): Client
+  createClient(apiKeyOrOpts: string | WorkerConfig): Client
+}
+
+const name = 'Bugsnag Web Worker'
+const url = 'https://github.com/bugsnag/bugsnag-js'
+// @ts-ignore
+const version = __BUGSNAG_NOTIFIER_VERSION__
+
+// extend the base config schema with some worker-specific options
+const schema = assign({}, baseConfig, workerConfig)
+
+type WorkerClient = Partial<ClientWithInternals> & {
+  _client: ClientWithInternals | null
+  createClient: (opts?: Config) => ClientWithInternals
+  start: (opts?: Config) => ClientWithInternals
+  isStarted: () => boolean
+}
+
+const notifier: WorkerClient = {
+  _client: null,
+  createClient: (opts) => {
+    // handle very simple use case where user supplies just the api key as a string
+    if (typeof opts === 'string') opts = { apiKey: opts }
+    if (!opts) opts = {} as unknown as Config
+
+    const internalPlugins = [
+      pluginBrowserDevice(navigator, null),
```

Comment:
As per `delivery-fetch` comment below maybe these plugins would be better off not assuming what environment they are running in